app/src/main/java/me/saket/dank/ui/submission/adapter/ImageWithMultipleVariants.kt

```diff
@@ -36,24 +41,33 @@ class ImageWithMultipleVariants private constructor(private val optionalRedditPr
       }
     }
 
-    // Reddit sends HTML-escaped URLs.
-    return Html.fromHtml(closestImage.url).toString()
+    return if (closestImage.width < minWidthChecked) null else {
+      // Reddit sends HTML-escaped URLs.
+      Html.fromHtml(closestImage.url).toString()
+    }
   }
 
-  fun findNearestFor(preferredWidth: Int, defaultValue: String): String {
+  fun findNearestFor(preferredWidth: Int): String {
+    return this.findNearestFor(preferredWidth, -1) ?:
+      throw NoSuchElementException("No reddit supplied images present")
+  }
+
+  fun findNearestFor(preferredWidth: Int, minWidth: Int, defaultValue: String): String {
     if (UrlParser.isGifUrl(defaultValue)) {
       throw AssertionError("Optimizing GIFs is an error: $defaultValue")
     }
 
-    return if (optionalRedditPreviews.isPresent) {
-      findNearestFor(preferredWidth)
-    } else {
-      defaultValue
-    }
+    return this.findNearestFor(preferredWidth, minWidth) ?: defaultValue
+  }
+
+  fun findNearestFor(preferredWidth: Int, defaultValue: String): String {
+    return findNearestFor(preferredWidth, -1, defaultValue)
   }
 
   companion object {
 
+    const val DEFAULT_VIEWER_MIN_WIDTH = 1200
```

Comment:
Maybe this should be lower?

Reply:
Perhaps something like width of screen, or maybe half?

Reply:
I've added some logs and checked several random posts - seems like width values are fixed, so I think there's no real need to calculate minWidth, just choose a res which won't be too blurry to look at on average screen. 1200 fits this case pretty well. Using something bigger will make data saver useless for, at least, i.redd,it which doesn't provide lowQualityUrl

```
[Preview 1]
res: 154 x 300
[Preview 1]
res: 154 x 154
res: 330 x 330
res: 500 x 500
res: 1200 x 1200
res: 1700 x 1700
res: 2070 x 2070
[Preview 1]
res: 154 x 121
res: 330 x 242
res: 500 x 360
res: 1200 x 740
res: 1700 x 1320
res: 2070 x 1452
[Preview 1]
res: 154 x 153
res: 330 x 326
res: 500 x 475
res: 1200 x 1173
```